django_tex/filters.py

```diff
@@ -1,20 +1,46 @@
 from django.utils.formats import localize_input
 from django.template.defaultfilters import register
 
+BACKSLASH_REPLACEMENT = "<:-!~#BACKSLASH#~!:->"
 
-def do_linebreaks(value):
+
+def do_linebreaks(value: str) -> str:
     return value.replace("\n", "\\\\\n")
 
 
-def do_latex_escape(value):
+def do_latex_escape(value: str) -> str:
+    """
+    Replace all LaTeX characters that could cause the latex compiler to fail
+    and at the same time try to display the character as intended from the user.
+
+    see also https://tex.stackexchange.com/questions/34580/escape-character-in-latex
+    """
+
     return (
-        value.replace("&", "\\&")
-        .replace("$", "\\$")
-        .replace("%", "\\%")
-        .replace("#", "\\#")
-        .replace("_", "\\_")
-        .replace("{", "\\{")
-        .replace("}", "\\}")
+        # Replace Backslashes so we dont' double replace backslashes
+        value.replace("\\", BACKSLASH_REPLACEMENT)
+        # If there was a space before the character we assume the user wants
+        # it also in the displayed LaTeX text
+        .replace(BACKSLASH_REPLACEMENT + " ", r"\textbackslash\space ")
+        # Otherwise we assume it doesn't matter. Note the space at the end is
+        # required that an additional text is not considered as part of the latex
+        # command
+        .replace(BACKSLASH_REPLACEMENT, r"\textbackslash ")
+        # Do the same for ~ and ^
+        .replace("~ ", r"\textasciitilde\space ")
+        .replace("~", r"\textasciitilde ")
+        .replace("^ ", r"\textasciicircum\space ")
+        .replace("^", r"\textasciicircum ")
+        .replace("&", r"\&")
+        .replace("$", r"\$")
+        .replace("%", r"\%")
+        .replace("#", r"\#")
+        .replace("_", r"\_")
+        .replace("{", r"\{")
+        .replace("}", r"\}")
+        # Replace vertical tab as this seems to be the only character from
+        # string.printable that can't be consumed from LaTeX
+        .replace(chr(11), "")
```

Comment:
This might be a matter of taste, but I think, it would be better to let LaTeX throw an error, if the input string contains characters that cannot be consumed, instead of silently removing bad characters.

Reply:
The whole idea was that end user (who input the variables) are never able to crash the LaTeX generation, no matter what is inputed.
What is use of the escape otherwise?

Reply:
I think, users should not put an illegal character in the input string in the first place. And if they do, they should be notified by an error message. For me, escaping means translating a legal character that has a certain meaning within the context of the prgramming language, i.e. a control character used by latex, into another legal character that has a different meaning.